cpp/include/raft/spatial/knn/detail/ivf_pq_search.cuh

```diff
@@ -979,7 +1007,7 @@ struct ivfpq_compute_similarity {
  *   3. split the query batch into smaller chunks, so that the device workspace
  *      is guaranteed to fit into GPU memory.
  */
-template <typename ScoreT, typename LutT, typename IdxT>
+template <typename T, typename ScoreT, typename LutT, typename IdxT>
```

Comment:
Rather than adding here an extra parameter, maybe it's better to define the `config` struct for half with the normalization constant == 1?

Reply:
@achirkin I need this template parameter to know the normalization constant, there is no other way to know what was the original dataset type afaik. Unless I missed something?

Reply:
What I mean is: `ScoreT` is `float` or `half`, but `T` could be `int8_t`. In the post-processing step, we'll convert `float` to `float` but using the normalization constant of `int8_t`

Reply:
Alternately if we want to limit template instantiations, we can pass the constant. What do you think?

Reply:
Ah, right, I forgot the ScoreT cannot be int8_t/uint8_t; never mind!

Reply:
> Alternately if we want to limit template instantiations, we can pass the constant. What do you think?

No, I don't like that this constant would sort of mean different things. We can move the postprocessing step for the scores out of the worker code completely though!

Reply:
> We can move the postprocessing step for the scores out of the worker code completely though!

Not really, `topk_dists` lives inside `ivfpq_search_worker` and uses `ScoreT`. Of course we could have two steps of post-processing but it wouldn't be efficient.